services/rollout-service/pkg/argo/argo.go

```diff
@@ -157,19 +160,21 @@ func (a ArgoAppProcessor) CreateOrUpdateApp(ctx context.Context, overview *api.G
 				}
 			}
 		} else {
-			validate := false
-			appToUpdate := CreateArgoApplication(overview, app, k.Environment)
+			appToUpdate := CreateArgoApplication(overview, app, env)
 			appUpdateRequest := &application.ApplicationUpdateRequest{
 				XXX_NoUnkeyedLiteral: struct{}{},
 				XXX_unrecognized:     nil,
 				XXX_sizecache:        0,
-				Validate:             ptr.Bool(validate),
+				Validate:             ptr.Bool(false),
 				Application:          appToUpdate,
 				Project:              ptr.FromString(appToUpdate.Spec.Project),
 			}
-			_, err := a.ApplicationClient.Update(ctx, appUpdateRequest)
-			if err != nil {
-				logger.FromContext(ctx).Error("updating application: "+appToUpdate.Name+",env "+env.Name, zap.Error(err))
+			//We have to exclude the unexported type isServerInferred. It is managed by Argo.
+			if !cmp.Equal(appUpdateRequest.Application.Spec, existingApp.Spec, cmp.AllowUnexported(v1alpha1.ApplicationSpec{}.Destination)) {
+				_, err := a.ApplicationClient.Update(ctx, appUpdateRequest)
```

Comment:
One thing that would be great here would be a tracing span around this whole method with the diff as annotation. Then it would be really easy to see how often apps are updated and why.

Reply:
This would be nice for all calls to the argo client (but that can be done separately).

Reply:
Agreed and added!

Reply:
I see a span, I don't see the the diff on the span.
I expected something like `span.setTag("argoDiff", diff)`

Reply:
Missed the push for the commit to add the diff, pushed